File: nodecore-cli/src/main/java/nodecore/cli/contracts/Shell.java
// VeriBlock NodeCore CLI
// Copyright 2017-2019 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package nodecore.cli.contracts;

import org.jline.utils.AttributedStyle;

import javax.net.ssl.SSLException;

public interface Shell {
    Result run();

    void initialize(String host);

    public void format(AttributedStyle style, int fColor, String str);

    ProtocolEndpointType type();

    void format(String fmt, Object... args);

    public boolean connect(ProtocolEndpoint endpoint, boolean save) throws SSLException, ConnectionFailedException;

    String passwordPrompt(String prompt);
}

File: altchain-pop-miner/miner/src/main/kotlin/org/veriblock/miners/pop/shell/Shell.kt
// VeriBlock Blockchain Project
// Copyright 2017-2018 VeriBlock, Inc
// Copyright 2018-2019 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package org.veriblock.miners.pop.shell

import org.veriblock.lite.NodeCoreLiteKit
import org.veriblock.miners.pop.Miner
import org.veriblock.miners.pop.service.PluginService
import org.veriblock.miners.pop.shell.commands.altchainCommands
import org.veriblock.miners.pop.shell.commands.configCommands
import org.veriblock.miners.pop.shell.commands.miningCommands
import org.veriblock.miners.pop.shell.commands.standardCommands
import org.veriblock.miners.pop.shell.commands.walletCommands
import org.veriblock.shell.Shell

fun Shell.configure(
    miner: Miner,
    nodeCoreLiteKit: NodeCoreLiteKit,
    pluginService: PluginService
) {
    standardCommands()
    configCommands()
    miningCommands(miner)
    walletCommands(miner)
    altchainCommands(nodeCoreLiteKit, pluginService)
}
